### PR TITLE
Parse HydraId and store cluster information

### PR DESCRIPTION
**Pull request by taymoork2**
Fixes [SPARK-206010](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-206010)

### CODE REVIEW

File: packages/node_modules/@webex/react-component-utils/src/uuid-utils.js [hunk 1]
```diff
@@ -34,24 +57,25 @@ export function getConvoIdFromUrl(convoUrl) {
...
     return {
-      id: string
+      id
     };
   }
 
   // Check for base 64 encoding
   try {
-    const idParts = deconstructHydraId(string);
+    const idParts = deconstructHydraId(id, cluster);
```
Comment (mwegman) [inline packages/node_modules/@webex/react-component-utils/src/uuid-utils.js:78]: `deconstructHydraId` doesn't take a second paramter

**Review by mwegman — CHANGES_REQUESTED** [review]

File: packages/node_modules/@webex/react-component-utils/src/uuid-utils.js [hunk 2]
```diff
@@ -72,27 +95,23 @@ export function validateAndDecodeId(string) {
...
- *
- * @export
- * @param {String} id
- * @returns {object} hydra
- * @returns {String} hydra.id
- * @returns {String} hydra.type
- * @returns {String} hydra.cluster
- */
-export {deconstructHydraId};
+export function constructHydraId(
```
Comment (adamweeks) [inline packages/node_modules/@webex/react-component-utils/src/uuid-utils.js:101]: `@webex/common` has a `constructHydraId`, can we just use that instead?

File: test/journeys/lib/hydra.js [hunk 3]
```diff
@@ -1,8 +1,27 @@
 import {base64} from '@webex/common';
```
Comment (adamweeks) [inline test/journeys/lib/hydra.js:1]: Can we use the construct and deconstruct from the sdk here?

File: packages/node_modules/@webex/widget-space/src/enhancers/setup.js [hunk 4]
```diff
@@ -79,17 +79,30 @@ export function setup(props, prevProps = {}) {
...
         }
-        if (destination.type === destinationTypes.SPACEID && !spaces.getIn(['byId', validateAndDecodeId(destination.id).id])) {
-          props.fetchSpace(sparkInstance, validateAndDecodeId(destination.id));
+        if (
+          destination.type === destinationTypes.SPACEID &&
+          !spaces.getIn(['byId', validateAndDecodeId(destination.id).id])
+        ) {
+          props.fetchSpace(
+            sparkInstance,
+            validateAndDecodeId(destination.id, destination.cluster)
```
Comment (adamweeks) [inline packages/node_modules/@webex/widget-space/src/enhancers/setup.js:104]: This function doesn't take a second param.

Comment (taymoork2) [reply packages/node_modules/@webex/widget-space/src/enhancers/setup.js:104]: This list of if blocks was prettier'd so i could read it
the second param was already there, but i can remove it

https://github.com/webex/react-widgets/blob/master/packages/node_modules/@webex/widget-space/src/enhancers/setup.js#L92

Comment (adamweeks) [reply packages/node_modules/@webex/widget-space/src/enhancers/setup.js:104]: I'm saying `validateAndDecodeId` doesn't take two params.

https://github.com/webex/react-widgets/blob/1310dae30e7a86f65b4a1b59e2e8d8a0a691f401/packages/node_modules/%40webex/react-component-utils/src/uuid-utils.js#L75

**Review by mwegman — APPROVED** [review]

**Pull request merged**